jung-algorithms/src/main/java/edu/uci/ics/jung/layout/algorithms/TreeLayoutAlgorithm.java

```diff
@@ -152,54 +158,28 @@ private int calculateDimensionX(LayoutModel<N> layoutModel, Collection<N> roots)
     return size;
   }
 
-  protected void setCurrentPositionFor(LayoutModel<N> layoutModel, N node) {
-    int width = layoutModel.getWidth();
-    int height = layoutModel.getHeight();
-    int x = (int) this.currentX;
-    int y = (int) this.currentY;
-    if (x < 0) {
-      width -= x;
-    }
+  private int calculateDimensionY(LayoutModel<N> layoutModel, N node) {
 
-    if (x >= width - distX) {
-      width = x + distX;
-    }
+    int size = 0;
+    int childrenNum = layoutModel.getGraph().successors(node).size();
 
-    if (y < 0) {
-      height -= y;
-    }
-    if (y >= height - distY) {
-      height = y + distY;
-    }
-    if (layoutModel.getWidth() < width || layoutModel.getHeight() < height) {
```

Comment:
previous version would recalculate layout size. The problem is that some nodes were already placed, so subsequent items would no longer line up nicely